File: app/src/main/java/ch/unstable/ost/views/lists/query/QueryHistoryAdapter.java
package ch.unstable.ost.views.lists.query;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.google.common.base.Preconditions;
import com.google.common.base.Verify;

import java.util.ArrayList;
import java.util.List;

import ch.unstable.ost.R;
import ch.unstable.ost.database.model.QueryHistory;


public class QueryHistoryAdapter extends RecyclerView.Adapter<QueryViewHolder> {

    private final View.OnClickListener mOnClickListener;
    private ArrayList<QueryHistory> mHistoryItems;

    public QueryHistoryAdapter(View.OnClickListener onClickListener) {
        mHistoryItems = new ArrayList<>();
        setHasStableIds(true);
        mOnClickListener = Preconditions.checkNotNull(onClickListener, "onClickListener");
    }

    @Override
    public QueryViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View itemView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_connection_query, viewGroup, false);
        return new QueryViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(final QueryViewHolder viewHolder, int i) {
        QueryHistory queryEntry = getQueryHistoryAt(i);
        QueryBinder.INSTANCE.bindQuery(queryEntry, viewHolder.getDate(), viewHolder.getFromAndTo());
        viewHolder.itemView.setTag(queryEntry);
        viewHolder.itemView.setClickable(true);
        viewHolder.itemView.setOnClickListener(mOnClickListener);
    }

    @NonNull
    public QueryHistory getQueryHistoryAt(int i) {
        return mHistoryItems.get(i);
    }

    @Override
    public long getItemId(int position) {
        long id = getQueryHistoryAt(position).getId();
        Verify.verify(id != 0, "id is 0");
        return id;
    }

    @Override
    public int getItemCount() {
        return mHistoryItems.size();
    }

    public void setEntries(List<QueryHistory> entries) {
        //noinspection ResultOfMethodCallIgnored
        Preconditions.checkNotNull(entries, "entries is null");
        mHistoryItems = new ArrayList<>(entries);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/ch/unstable/ost/views/lists/query/QueryHistoryAdapter.kt
package ch.unstable.ost.views.lists.query

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import ch.unstable.ost.R
import ch.unstable.ost.database.model.QueryHistory
import ch.unstable.ost.views.lists.query.QueryBinder.bindQuery
import java.util.*

class QueryHistoryAdapter(private val mOnClickListener: View.OnClickListener) : RecyclerView.Adapter<QueryViewHolder>() {
    private var mHistoryItems: ArrayList<QueryHistory>
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): QueryViewHolder {
        val itemView = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_connection_query, viewGroup, false)
        return QueryViewHolder(itemView)
    }

    override fun onBindViewHolder(viewHolder: QueryViewHolder, i: Int) {
        val queryEntry = getQueryHistoryAt(i)
        bindQuery(queryEntry, viewHolder.date, viewHolder.fromAndTo)
        viewHolder.itemView.tag = queryEntry
        viewHolder.itemView.isClickable = true
        viewHolder.itemView.setOnClickListener(mOnClickListener)
    }

    fun getQueryHistoryAt(i: Int): QueryHistory {
        return mHistoryItems[i]
    }

    override fun getItemId(position: Int): Long {
        val id = getQueryHistoryAt(position).id
        require(id == 0L) { "id is 0" }
        return id
    }

    override fun getItemCount(): Int {
        return mHistoryItems.size
    }

    fun setEntries(entries: List<QueryHistory>) {
        mHistoryItems = ArrayList(entries)
        notifyDataSetChanged()
    }

    init {
        mHistoryItems = ArrayList()
        setHasStableIds(true)
    }
}